### DISCUSSION TITLE
Getting an Error when i used a metabolite Identifier that contains a number

### DISCUSSION THREAD

**Discussion by Rahmaabubakr** [opencobra/cobrapy#1358] [Q&A] [answered]
Hello,

when i used this 
model.metabolites.5mthf_LSQBKTc_RSQBKT.summary()

and the metabolite identifier contains a number, i get this error:
Cell In[72], line 1
    model.metabolites.5mthf_LSQBKTc_RSQBKT.summary()
                      ^
SyntaxError: invalid decimal literal

it works fine with other metabolites that their identifier don't contain a number, however, it occurs with all id with numbers, what should i do? 

Thank you very much in advance

**Comment by matthiaskoenig** [comment, marked as answer]
Hi @Rahmaabubakr,
will not work with numbers. It is just a very bad idea to have identifiers starting with numbers.
These are not valid SBML identifiers nor valid python variable names. So they will create a ton of issues everywhere in your workflows.
There is a lot of discussion about this in the issues, see for instance: https://github.com/opencobra/cobrapy/issues/883

My personal recommendation: replace the identifiers of these metabolites.
Best Matthias

**Reply by Rahmaabubakr** [reply]
Thank you very much.
So, should i Edit the model and work from the Edited version? 
So, how can i open a file from the computer not from biomodels?

**Reply by matthiaskoenig** [reply]
You can download the SBML from biomodels. You can select the SBML file from Biomodels.
First I would check that the model is valid SBML, i.e. run it once through a validator: https://sbml.org/validator_servlet/ 
I.e. the identifier problem could be a hint for more things to be wrong in the model.

If you are sure the model is valid you could replace the ids either manually (which I would not recommend) or based on a python script.
